examples/s3express/main.tf

```diff
@@ -0,0 +1,156 @@
+    s3express = {
+      service         = "s3express"
+      service_type    = "Gateway"
+      route_table_ids = module.vpc.private_route_table_ids
+      tags            = { Name = local.name }
+    }
+  }
+}
+
+# Creating a security group to allow things in the VPC
+# to connect to the WarpStream Agents.
+resource "aws_security_group" "warpstream-connect" {
+  name        = "${local.name}-connect"
+  description = "Allow things the VPC to connect to the WarpStream Kafka Port"
+  vpc_id      = module.vpc.vpc_id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "allow_agent_to_agent_http" {
+  security_group_id = aws_security_group.warpstream-connect.id
+  description       = "Allow Server to Agent Kafka Communication"
+
+  ip_protocol = "tcp"
+  from_port   = 9092
+  to_port     = 9092
+  cidr_ipv4   = "10.0.0.0/16"
+}
+
+# Store the WarpStream Agent Key in AWS Secret Manager
+resource "random_id" "warpstream_agent_key" {
+  byte_length = 4
+}
+
+resource "aws_secretsmanager_secret" "warpstream_agent_key" {
+  name = "${local.name}-agent-key-${random_id.warpstream_agent_key.hex}"
+}
+
+resource "aws_secretsmanager_secret_version" "warpstream_agent_key" {
+  secret_id     = aws_secretsmanager_secret.warpstream_agent_key.id
+  secret_string = var.warpstream_agent_key
+}
+
+module "warpstream" {
+  source = "../.."
+
+  depends_on = [module.vpc, module.endpoints]
+
+  cluster_name         = local.name
+  control_plane_region = local.region
+
+  warpstream_virtual_cluster_id           = var.warpstream_virtual_cluster_id
+  warpstream_agent_key_secret_manager_arn = aws_secretsmanager_secret_version.warpstream_agent_key.arn
+
+  # We recommend network optimized instances with a minimum of 4 vCPUs and 16gb Memory to get the best performance.
+  # We have also tested with Graviton3+ and found decent performance. 
+  # The ECS tasks assume 1:4 vCPU to Memory ratio with 1 core and 4gb of ram left to the host OS.
+  ec2_instance_type = "m6in.xlarge"
+
+  # Add the default VPC security group to the ECS EC2 instances.
+  ec2_instance_security_group_ids = [module.vpc.default_security_group_id]
+
+  # List of subnet IDs to launch ecs ec2 VMs in. 
+  # Subnets automatically determine which availability zones the ec2 group will reside.
+  ec2_vpc_zone_identifier = module.vpc.private_subnets
+
+  # The VPC that the warpstream ECS service runs on
+  ecs_service_vpc_id = module.vpc.vpc_id
+
+  # List of subnet IDs to launch the ecs service in. 
+  # The subnets can be different then the ec2_vpc_zone_identifier
+  # Subnets automatically determine which availability zones the ecs service will reside.
+  ecs_subnet_ids = module.vpc.private_subnets
+
+  # Specifying the security group to allow things in the VPC to connect to WarpStream agents.
+  ecs_service_additional_security_group_ids = [aws_security_group.warpstream-connect.id]
+
+  bucket_names           = aws_s3_directory_bucket.s3_express_buckets[*].bucket
```

Comment:
Seems like the module could just create the buckets so people dont have to manage / think about this at all?

Reply:
We dont want people using shared buckets anyways and S3 allows up to 1million s3 buckets now